### PR TITLE
Fix issue 135

### PR DESCRIPTION
**Pull request by andriish**
Hi @lyskov ,

this is a fix for issue #135 .
Basically that is a backport of LLVM6+ bugfix. 

Best regards,

Andrii

### CODE REVIEW

File: source/enum.cpp [hunk 1]
```diff
@@ -56,6 +56,27 @@ bool is_bindable(EnumDecl const *E)
...
+	while (Ctx && isa<NamedDecl>(Ctx)) {
+		Contexts.push_back(Ctx);
+		Ctx = Ctx->getParent();
+	}
+	for (const DeclContext *DC : reverse(Contexts)) {
+		if (const auto *ED = dyn_cast<EnumDecl>(DC)) {
+			if ( ED->isScoped() ) {OS<<*ED; OS<<"::";} else continue;
+		} 
+		else 
+			{ OS << *cast<NamedDecl>(DC);  OS<<"::";}
```
Comment (lyskov) [inline source/enum.cpp:75]: please fix indentation and bracket  new lines

**Review by lyskov — COMMENTED** [review]
@andriish please see a few minor comments below. I am curious, why support for LLVM5 is needed, what OS's have it?

File: source/enum.cpp [hunk 2]
```diff
@@ -56,6 +56,27 @@ bool is_bindable(EnumDecl const *E)
 	return true;
 }
 
+// This function takes care about the LLVM/Clang bug which was fixed in LLVM6/Clang6.
+// The body of the function is a backport from LLVM6.
+static std::string getQualifiedNameAsStringLLVM5Fix( NamedDecl const *E) {
```
[the remark this answers is not in the document]
Comment (andriish) [reply source/enum.cpp:61]: OK.

File: source/enum.cpp (hunk 1 above)
Comment (andriish) [reply source/enum.cpp:75]: ok

File: source/enum.cpp [hunk 3]
```diff
@@ -56,6 +56,27 @@ bool is_bindable(EnumDecl const *E)
...
+	std::string correct;
+	llvm::raw_string_ostream OS(correct);
+	const DeclContext *Ctx = E->getDeclContext();
+	SmallVector<const DeclContext *, 10> Contexts;
```
[the remark this answers is not in the document]
Comment (andriish) [reply source/enum.cpp:65]: > Is it max number of elements before heap allocation is used?
Very likely . That was 8 in the original code. 

> Is it safe to push more elements in such vector?
I bet it is .  The same is used in LLVM.

**Comment by andriish** [comment]
> what OS's have it?  
Basically that is CI -- Fedora26/27.   Other distros do have it as well, of course.
It is not that the results are "wrong", it is about uniformity.

**Review by lyskov — APPROVED** [review]
LGTM! Thank you @andriish !

**Pull request merged**